openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -124,9 +156,33 @@ class AccessTokenExchangeView(_DispatchingView):
     """
     dot_view = auth_exchange_views.DOTAccessTokenExchangeView
 
+    def dispatch(self, request, *args, **kwargs):
+        response = super().dispatch(request, *args, **kwargs)
+        token_type = _get_token_type(request)
+
+        if response.status_code == 200 and token_type == 'jwt':
+            response.data = self._get_jwt_data_from_access_token_data(request, response)
+
+        return response
+
+    def _get_jwt_data_from_access_token_data(self, request, response):
+        """
+        Gets the JWT response data from the opaque token response data.
+
+        Includes the JWT token and token type in the response.
+        """
+        opaque_token_dict = response.data
+        jwt_token_dict = _get_jwt_dict_from_opaque_access_token_dict(
+            opaque_token_dict, self.get_adapter(request)
+        )
+        return jwt_token_dict
+
 
 class RevokeTokenView(_DispatchingView):
     """
     Dispatch to the RevokeTokenView of django-oauth-toolkit
+
+    Note: JWT access tokens are non-revocable, but you could still revoke
```

Comment:
In another thread, Jawad answered:
> Mobile calls revoke_token only when a user clicks logout. And non-revocable nature of jwt doesn't create problem for us.
> Yes revoke_token revokes refresh_token. Yes, on mobile side we will discard jwt after logout call and it will expire after 1 hour.